share/completions/history.fish

```diff
@@ -1,10 +1,33 @@
-complete -c history -r -l prefix --description "Match items starting with prefix"
-complete -c history -r -l contains --description "Match items containing string"
-complete -c history -l search -s s --description "Prints commands from history matching query"
-complete -c history -l delete -s d --description "Deletes commands from history matching query"
-complete -c history -l clear --description "Clears history file"
-complete -c history -l merge -s m --description "Incorporate history changes from other sessions"
-complete -c history -l exact -s e --description "Match items in the history that are identicial"
+set -g __fish_history_all_commands search delete save merge clear
```

Comment:
Given the scoping rules it will be global even without the `-g`. I vaguely recall a discussion about enhancing this so that there is a "file" scope analogous to python's module scope. The other option is to wrap all of this inside

```
function __fish_history_completions
...
end
__fish_history_completions
functions -e __fish_history_completions
```

That allows using `set -l`. The fix seems worse than simply making it a private global var.